### ISSUE TITLE
SearXNG service does not work with default options

### ISSUE THREAD

**Issue by xieve** [NixOS/nixpkgs#292637]
### Describe the bug
`services.searx` fails to start with the default settings. This is the relevant part of the output of `nixos-rebuild`:

```
starting the following units: searx-init.service
warning: the following units failed: searx.service

× searx.service - Searx server, the meta search engine.
     Loaded: loaded (/etc/systemd/system/searx.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Fri 2024-03-01 19:16:49 CET; 69ms ago
   Duration: 5ms
    Process: 163773 ExecStart=/nix/store/av5mvbmsr8k35r5m98pqlzi4z9qdh3hn-searxng-0-unstable-2024-02-24/bin/searx-run (code=exited, status=203/EXEC)
   Main PID: 163773 (code=exited, status=203/EXEC)
         IP: 0B in, 0B out
        CPU: 1ms

Mär 01 19:16:49 thegreatbelow systemd[1]: Started Searx server, the meta search engine..
Mär 01 19:16:49 thegreatbelow systemd[1]: searx.service: Main process exited, code=exited, status=203/EXEC
Mär 01 19:16:49 thegreatbelow systemd[1]: searx.service: Failed with result 'exit-code'.
```

### Steps To Reproduce
1. Add this to system config: 
```Nix
services.searx.enable = true;
```
2. Rebuild

### Expected behavior
`services.searx` should start without any additional configuration. This would match the behavior of setting up SearXNG manually.

### Additional context

```ini
# /etc/systemd/system/searx.service

[Unit]
After=searx-init.service
Description=Searx server, the meta search engine.
Requires=searx-init.service

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/1ri11hgwr2mxzf6wadcwc46rijk52r2q-glibc-locales-2.38-44/lib/locale/locale-archive"
Environment="PATH=/nix/store/x1xcjw5628crkk1pwr12y7nwbzkc3969-coreutils-9.4/bin:/nix/store/i6y16f2jzcv1g1k12qdkislh3yqk2rl7-findutils-4.9.0/bin:/nix/store/11b3chszacfr9liy829xqknzp3q88iji-gnugrep-3.11/bin:/nix/store/y1y3rml47qnh0giqd32mj07qxxqy13qg-gnused-4.9/bin:/nix/store/s5dx34869kl4fd5p913j5mis32c7f98k-systemd-255.2/bin:/nix/store/x1xcjw5628crkk1pwr12y7nwbzkc3969-coreutils-9.4/sbin:/nix/store/i6y16f2jzcv1g1k12qdkislh3yqk2rl7-findutils-4.9.0/sbin:/nix/store/11b3chszacfr9liy829xqknzp3q88iji-gnugrep-3.11/sbin:/nix/store/y1y3rml47qnh0giqd32mj07qxxqy13qg-gnused-4.9/sbin:/nix/store/s5dx34869kl4fd5p913j5mis32c7f98k-systemd-255.2/sbin"
Environment="SEARXNG_SETTINGS_PATH=/run/searx/settings.yml"
Environment="SEARX_SETTINGS_PATH=/run/searx/settings.yml"
Environment="TZDIR=/nix/store/lb1z8nckdwx4kmwy19znh1bl1p63wxhv-tzdata-2024a/share/zoneinfo"
ExecStart=/nix/store/av5mvbmsr8k35r5m98pqlzi4z9qdh3hn-searxng-0-unstable-2024-02-24/bin/searx-run
Group=searx
User=searx

[Install]
WantedBy=network.target multi-user.target
```
```console
$ ls /nix/store/av5mvbmsr8k35r5m98pqlzi4z9qdh3hn-searxng-0-unstable-2024-02-24/bin/
searxng-checker  searxng-run
```
It seems like the main binary was renamed from `searx-run` to `searxng-run` in the fork (SearXNG), which is the default package used by `services.searx` (very counterintuitively, especially since SearX has been removed from nixpkgs entirely).

If you set `services.searx.runInUwsgi = true`, this bug does not occur, as the main executable is not called by any service.

### Notify maintainers

@rnhmjoj @999eagle 

### Metadata

```console
$ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240226.13aff9b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by chewblacka** [comment]
I think this was addressed / fixed here (only merged yesterday):
https://github.com/NixOS/nixpkgs/pull/292185

**Comment by xieve** [comment]
> I think this was addressed / fixed here (only merged yesterday): #292185

My bad, I only searched existing issues for "SearXNG".
